emission/net/api/cfc_webapp.py

```diff
@@ -403,8 +403,14 @@ def getPublicData():
 
   data_list = map(lambda q: list(edb.get_timeseries_db().find(q).sort("metadata.write_ts")), user_queries)
 
-  logging.debug("public data retrieved")
-
+  num_entries = map(lambda q: len(q), data_list)
```

Comment:
I think that we should do this check sooner. Basically, 
`list(edb.get_timeseries_db().find(q).sort("metadata.write_ts"))` pulls all the data into memory already.

Instead, you should first do a call to `find().count()` which will give you a count *before* reading data into memory, and only read the full data once you know that it is safe.